async-nats/src/jetstream/consumer/push.rs

```diff
@@ -255,3 +255,218 @@ impl IntoConsumerConfig for &Config {
+    pub filter_subject: String,
+    /// Whether messages are sent as quickly as possible or at the rate of receipt
+    pub replay_policy: ReplayPolicy,
+    /// The rate of message delivery in bits per second
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub rate_limit: u64,
+    /// What percentage of acknowledgements should be samples for observability, 0-100
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub sample_frequency: u8,
+    /// Only deliver headers without payloads.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub headers_only: bool,
+    /// Allows for a variety of options that determine how this consumer will receive messages
+    #[serde(flatten)]
+    pub deliver_policy: DeliverPolicy,
+    /// The maximum number of waiting consumers.
+    #[serde(default, skip_serializing_if = "is_default")]
+    pub max_waiting: i64,
+}
+
+impl FromConsumer for OrderedConfig {
+    fn try_from_consumer_config(config: crate::jetstream::consumer::Config) -> Result<Self, Error>
+    where
+        Self: Sized,
+    {
+        if config.deliver_subject.is_none() {
+            return Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "push consumer must have delivery subject",
+            )));
+        }
+        Ok(OrderedConfig {
+            deliver_subject: config.deliver_subject.unwrap(),
+            description: config.description,
+            filter_subject: config.filter_subject,
+            replay_policy: config.replay_policy,
+            rate_limit: config.rate_limit,
+            sample_frequency: config.sample_frequency,
+            headers_only: config.headers_only,
+            deliver_policy: config.deliver_policy,
+            max_waiting: config.max_waiting,
+        })
+    }
+}
+
+impl IntoConsumerConfig for OrderedConfig {
+    fn into_consumer_config(self) -> super::Config {
+        jetstream::consumer::Config {
+            deliver_subject: Some(self.deliver_subject),
+            durable_name: None,
+            description: self.description,
+            deliver_group: None,
+            deliver_policy: self.deliver_policy,
+            ack_policy: AckPolicy::None,
+            ack_wait: Duration::from_secs(60 * 60 * 22),
+            max_deliver: 1,
+            filter_subject: self.filter_subject,
+            replay_policy: self.replay_policy,
+            rate_limit: self.rate_limit,
+            sample_frequency: self.sample_frequency,
+            max_waiting: self.max_waiting,
+            max_ack_pending: 0,
+            headers_only: self.headers_only,
+            flow_control: true,
+            idle_heartbeat: Duration::from_secs(5),
+            max_batch: 0,
+            max_expires: Duration::default(),
+            inactive_threshold: Duration::from_secs(30),
+        }
+    }
+}
+
+impl Consumer<OrderedConfig> {
+    pub async fn ordered<'a>(self) -> Result<Ordered<'a>, Error> {
```

Comment:
The consumer is ordered, so just messages?

```suggestion
    pub async fn messages<'a>(self) -> Result<Messages<'a>, Error> {
```